### PR TITLE
build(socket.io, socket.io-client): upgrade to common socket.io version

### PR DESCRIPTION
**Pull request by outSH**
Some packages use socket.io and socket.io-client V2, while other use V4.
In order for these components to communicate, we must use common
socket.io version in all cactus packages (V4.1.3). This commit introduce
some other related changes, like merging socketio unit tests to common
setup (to assert changes are correct) and fixes some strict-flag
warnings and minor bugs. It also updates python packages requirements
and updates readme when it was necessary.

Closes: #1679
Signed-off-by: Michal Bajer <michal.bajer@fujitsu.com>

Additional changes in **cactus-cmd-socketio-server**:
- Removed local jest config, test will be run from the root dir now.
- Fixed several strict flag warnings from verifier component.
- Removed nohoist pragma, changed package to public and added license info.
- Added monitor_error handle to fix few unit tests (cherry-picked from #1660)
- Used common socketio test helpers (cherry-picked from #1660)

**Depends on #1670**

### CODE REVIEW

**Comment by petermetz** [comment]
@outSH Thank you very much for addressing this so quickly, I'm very grateful for the quick turnaround!
Right now the PR has a single commit that includes the full diff of the PR that this one is dependent on, there's two options we can do to make it easy to review:
1. Seperate out the commits so that the diffs can be reviewed separately
2. Or just wait until #1670 gets merged and then after a rebase this will have the correct diff to be reviewed.

I'm fine either way. Regarless of which one you choose, this is looking great at first glance.

File: packages/cactus-common/src/main/typescript/socketio-test-setup-helpers.ts
Comment (petermetz) [inline packages/cactus-common/src/main/typescript/socketio-test-setup-helpers.ts:11]: @outSH Apologies for the nit picking: Could you move this file to the `cactus-test-tooling` package instead of `cactus-common`? 
Reasoning: If it's specifically meant for testing then I'd rather not have it increase the bundle size of the common package. I know that the `Servers` utility class looks similarly out of place, but that one was originally meant to be the utility that the `cmd-api-server` uses when doing graceful HTTP server shutdowns so that's why that one had to be put here in the `cactus-common` package. The fact that 99% of its use-cases for the `Servers` class is actually in the test automation is just a happy coincidence.

**Review by petermetz — CHANGES_REQUESTED** [review]

**Comment by outSH** [comment]
Depends on #1660

File: packages/cactus-common/src/main/typescript/socketio-test-setup-helpers.ts
Comment (outSH) [reply packages/cactus-common/src/main/typescript/socketio-test-setup-helpers.ts]: @petermetz Sure, no problem, any tips are welcomed :) I was in fact mimicking `Servers` class and thus put it in the same folder, but I agree with your reasoning. I've moved `socketio-test-setup-helpers` to `cactus-test-tooling` both in this PR, and in #1660 (from which this change originated).

**Comment by outSH** [comment]
@petermetz 
I've changed this PR to draft, after #1660 and #1670 are merged I will double check and do some test to ensure everything is fine, and we'll proceed :) In the future I'll probably just merge dependent commits to clarify the diffs

**Comment by petermetz** [comment]
@outSH Understood, sounds like a plan! :-)

**Comment by outSH** [comment]
Dependent PR were merged, I've done tests after rebasing to main and all seems fine.

@petermetz @izuru0 Please review the changes :)

**Review by izuru0 — APPROVED** [review]
LGTM

**Review by petermetz — APPROVED** [review]
@outSH LGTM, thank you very much.

**Pull request merged**
